### PR TITLE
Obstacle avoidance 

### PR DESCRIPTION
**Pull request by Valkama1**
Implemented ultrasonic sensor and steering to the sketch.

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -1,9 +1,65 @@
...
-  // put your setup code here, to run once:
+    Serial.begin(9600);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
-}
+    handleInput();
+
+    if (front.getDistance() < 50 && front.getDistance() != 0)
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino]: What happens if the first `getDistance` returns `0` and the second one `100`? The car will stop despite not having detected a close-enough obstacle.

**Review by platisd — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Valkama1) [reply arduino/smartcar/smartcar.ino]: Yeah I didn't consider that
Would it be a good idea to instead store a variable holding 1 getDistance value and compare that instead?
That way the same value should always be used in both comparisons

**Comment by Valkama1** [comment]
Okay now everything should work.

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (platisd) [reply arduino/smartcar/smartcar.ino]: :+1: :+1:

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -1,9 +1,71 @@
...
-  // put your setup code here, to run once:
+    Serial.begin(9600);
 }
 
 void loop() {
-  // put your main code here, to run repeatedly:
-}
+    handleInput();
+
+    auto temp = front.getDistance(); //Temp variable to compare in order to decide whether to stop or not
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:61]: Not a good variable name but it gets the job done I guess

**Review by platisd — APPROVED** [review]

**Pull request merged**